### ISSUE TITLE
Import\export *.cs file

### ISSUE THREAD

**Issue by holgworknote** [gbaychev/NClass#5]
Guess that its must-have feature for prototyping app like NClass.

**Comment by V0v1kkk** [comment]
Давно у меня в голове витает мысль переписать парсинг\генерацию C# на Roslyn'е.
Тогда и валидацию конструкций лучше делать с его помощью (вытекают глаза от такого количества регулярок + проблемы с поддержкой новых фитч языка).
Но у идеи есть и минус - придется отказаться от поддержки Java.

**Comment by gbaychev** [comment]
Hello,
1. there is already an option to export .cs/class file, even to generate a project.
2. cs file import with roslyn is not a bad idea - you may do that as a plugin, not in the core
3. java support won't be dropped.
4. If both of you don't translate your paragraphs in English and don't write in plain English from now on, I'm closing the issue and blocking you. Seriously, there is not going to be first or second strike, if I see another non-English sentence, I'm doing this.

Have a merry Christmas.

**Comment by holgworknote** [comment]
1) > there is already an option to export .cs/class file, even to generate a project.
I found this option, but there is still no option to generate NCP from project

2) Okay, I took a note, there is no need for extreme measures.

**Comment by gbaychev** [comment]
Hello and merry Christmas,
> I found this option, but there is still no option to generate NCP from project

You cannot import .cs files directly, yes, this is correct. What you can do is to compile your .cs into an assembly and then create a new project, new c# diagram and import the assembly:

![import](https://user-images.githubusercontent.com/3358056/34342975-dcd03380-e9c1-11e7-9606-4d1f3ab9a41f.png)

You can use this as a workaround for awhile, importing .cs files directly via roslyn is not a bad idea and can be implemented as a plugin in the future.

> Okay, I took a note, there is no need for extreme measures.

Okay, I'm sorry if I overreacted a bit. Simply this was not the first time I'm dealing with Russian speakers and usually they show a total disregard to what language the other side speaks.

Have a nice day/evening.
